benchmarking/src/benchmarking.ts

```diff
@@ -0,0 +1,127 @@
+import nodeTimesPromises from "node:timers/promises";
+import nodeFsPromises from "node:fs/promises";
+import nodePath from "node:path";
+
+export type FetchBenchmark = {
+  callDurationsMs: number[];
+  averageMs: number;
+};
+
+export type BenchmarkingResults = {
+  name: string;
+  path: string;
+  fetchBenchmark: FetchBenchmark;
+}[];
+
+/**
+ * Benchmarks the response time of an application end-to-end by:
+ *  - building the application
+ *  - deploying it
+ *  - and fetching from it (multiple times)
+ *
+ * @param options.build function implementing how the application is to be built
+ * @param options.deploy function implementing how the application is deployed (returning the url of the deployment)
+ * @param options.fetch function indicating how to fetch from the application (in case a specific route needs to be hit, cookies need to be applied, etc...)
+ * @returns the benchmarking results for the application
+ */
+export async function benchmarkApplicationResponseTime({
+  build,
+  deploy,
+  fetch,
+}: {
+  build: () => Promise<void>;
+  deploy: () => Promise<string>;
+  fetch: (deploymentUrl: string) => Promise<Response>;
+}): Promise<FetchBenchmark> {
+  await build();
+  const deploymentUrl = await deploy();
+  return benchmarkFetch(deploymentUrl, { fetch });
+}
+
+type BenchmarkFetchOptions = {
+  numberOfCalls?: number;
```

Comment:
Is "calls" really what we want? `numberOfIterations` ?

Reply:
yes `calls` never sit too well with me either 😕 

I am not sure if `iterations` is much better, but I'm happy with it 🙂